### DISCUSSION TITLE
Capacity remaining, add decimal points?

### DISCUSSION THREAD

**Discussion by Weezmo** [syssi/esphome-jk-bms#389] [Q&A] [answered]
Hi, how would i go about adding a decimal point or 2 to the esp code? line 512 in 
/components/jk_bms/sensor.py
?

**Comment by syssi** [comment]
This doesn't make sense. The measurement (state of charge in percent) of the BMS doesn't provide any decimal places:

https://github.com/syssi/esphome-jk-bms/blob/main/components/jk_bms/jk_bms.cpp#L153-L155

**Reply by syssi** [reply]
If you want two decimal places nevertheless which will be always `.00` you could extend the YAML configuration of the sensor by:

``` 
    capacity_remaining:
      name: "${name} capacity remaining"
      accuracy_decimals: 2
```

**Reply by Weezmo** [reply]
Okay, because it has a decimal point showing in the iOS app atleast
![IMG_4841](https://github.com/syssi/esphome-jk-bms/assets/128608360/2edd613d-9008-4e82-b44f-03c9c565b503)
But i guess it Will be forking a copy and using that then?


Thaaank you :)

**Reply by syssi** [reply]
The naming of the sensors at the project here are a bit confusing unfortunately.

The `capacity_remaining` sensor is labeled `Remain Battery: 99%` at the android app.

You are asking about the decimal places of the `Remain capacity` (Android app). This value is available as `capacity_remaining_derived`. You should increase the number of decimal places here:

```
    capacity_remaining_derived:
      name: "${name} capacity remaining derived"
      accuracy_decimals: 2
```

**Reply by Weezmo** [reply]
Im not sure if i misunderstood you
but i installed another of my esp32 with the code you made after you made an update to it, but it seems to still not send that extra decimal to homeassistant

![Capture2](https://github.com/syssi/esphome-jk-bms/assets/128608360/820772d6-eebb-4fdb-9457-919bf20ed541)

**Reply by syssi** [reply]
Oh, you mentioned the `jk_bms` (UART-TTL) component above but you are using the `jk_bms_ble` (BLE) implementation. So lets start from scratch. :-)

https://github.com/syssi/esphome-jk-bms/blob/main/components/jk_bms_ble/jk_bms_ble.cpp#L634-L635

The resolution is pretty good here. 3 decimal points. Let's check the default configuration at the implementation:

https://github.com/syssi/esphome-jk-bms/blob/main/components/jk_bms_ble/sensor.py#L641

I will increase this setting to `3` because the better resolution is available.

**Reply by syssi** [reply, marked as answer]
I've pushed a change to `main`. Please recompile & flash your setup and check the `capacity remaining` sensor another time.

**Reply by Weezmo** [reply]
Perfect that did it, Thank you
